File: passwordGenerator/pass.java
import java.security.SecureRandom;

public class PasswordGenerator {
    public static String generatePassword(int length) {
        String characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_=+";
        SecureRandom random = new SecureRandom();
        StringBuilder password = new StringBuilder();

        for (int i = 0; i < length; i++) {
            int index = random.nextInt(characters.length());
            password.append(characters.charAt(index));
        }

        return password.toString();
    }

    public static void main(String[] args) {
        int length = 12;
        String password = generatePassword(length);
        System.out.println(password);
    }
}

File: passwordGenerator/pass.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <string>

std::string generatePassword(int length) {
    std::string characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_=+";
    std::string password;

    std::srand(std::time(0)); // Seed the random number generator

    for (int i = 0; i < length; i++) {
        int index = std::rand() % characters.length();
        password.push_back(characters[index]);
    }

    return password;
}

int main() {
    int length = 12;
    std::string password = generatePassword(length);
    std::cout << password << std::endl;
    return 0;
}
